packages/@lwc/engine/dom/src/polyfills/aria-properties/polyfill.ts

```diff
@@ -6,8 +6,6 @@
  */
 import { hasOwnProperty, AriaPropNameToAttrNameMap } from '@lwc/shared';
 
-import { setAttribute, removeAttribute, getAttribute, hasAttribute } from '../../env/element';
```

Comment:
Theoretically, the only case where those APIs can be messed up is in the context of Lightning Out, otherwise, Locker prevents this. In the case of LO, even if the APIs are cached it will be too late since the LWC code will run after the page is loaded and there is high probability damage has already been done.

Reply:
In @lwc/engine, we have slowly moved away from caching browser methods to accessing them via the dom elements. This started with the global patching. I think this change is fine, but do expect there might be cases where the prototypes have been dirtied by downstream changes. 
Locker does not protect us always:
 1. Locker for aura is at a component level and there are a decent number of old aura components that run without locker.
 2. Communities runs without locker and their customers load third party libraries often.

Reply:
@ravijayaramappa I believe that we've only been invoking methods directly in places where we need shadow semantics defined by our synthetic shadow polyfill--at least that's how I've been making the decision of whether or not to cache a method.